FRENCH/src/ch07-02-defining-modules-to-control-scope-and-privacy.md

```diff
@@ -0,0 +1,249 @@
+pourront plus facilement trouver les définitions dont ils ont besoin car ils
+peuvent parcourir le code en fonction des groupes plutôt que d'avoir à lire
+toutes les définitions. Les développeurs qui veulent rajouter des nouvelles
+fonctionnalités à ce code sauront maintenant où placer le code tout en gardant
+le programme organisé.
+
+<!--
+Earlier, we mentioned that *src/main.rs* and *src/lib.rs* are called crate
+roots. The reason for their name is that the contents of either of these two
+files form a module named `crate` at the root of the crate’s module structure,
+known as the *module tree*.
+-->
+
+Précédemment, nous avons dit que *src/main.rs* et *src/lib.rs* étaient des
+racines de crates. Nous les appelons ainsi car le contenu de chacun de ces
+deux fichiers constituent un module qui s'appelle `crate` à la racine de la
+structure du module de cette crate, aussi appelée *l'arborescence du module*.
+
+<!--
+Listing 7-2 shows the module tree for the structure in Listing 7-1.
+-->
+
+L'encart 7-2 présente l'arborescence du module pour la structure de
+l'encart 7-1.
+
+<!--
+```text
+crate
+ └── front_of_house
+     ├── hosting
+     │   ├── add_to_waitlist
+     │   └── seat_at_table
+     └── serving
+         ├── take_order
+         ├── serve_order
+         └── take_payment
+```
+-->
+
+```text
+crate
+ └── salle_a_manger
+     ├── accueil
+     │   ├── ajouter_a_la_liste_attente
+     │   └── installer_a_une_table
+     └── service
+         ├── prendre_commande
+         ├── servir_commande
+         └── encaisser
+```
+
+<!--
+<span class="caption">Listing 7-2: The module tree for the code in Listing
+7-1</span>
+-->
+
+<span class="caption">Encart 7-2 : L'arborescence du module pour le code de
+l'encart 7-1</span>
+
+<!--
+This tree shows how some of the modules nest inside one another (for example,
+`hosting` nests inside `front_of_house`). The tree also shows that some modules
+are *siblings* to each other, meaning they’re defined in the same module
+(`hosting` and `serving` are defined within `front_of_house`). To continue the
+family metaphor, if module A is contained inside module B, we say that module A
+is the *child* of module B and that module B is the *parent* of module A.
+Notice that the entire module tree is rooted under the implicit module named
+`crate`.
+-->
+
+Cette arborescence comment les modules sont imbriqués entre eux (par exemple,
+`accueil` est imbriqué dans `salle_a_manger`). L'arborescence montre aussi que
+certains modules sont les *frères* d'autres modules, ce qui veut dire qu'ils
+sont définis dans le même module (`accueil` et `service` sont définis dans
+`salle_a_manger`). Pour prolonger la métaphore familiale, si le module A est
+compris dans un module B, on dit que le module A est *l'enfant* du module B et
+que ce module B est le *parent* du module A. Notez aussi que tout le module est
+lié au module implicite qui s'appelle `crate`.
```

Comment:
What about:
```MarkDown
tous les modules sont les descendents du module implicite `crate`.
```

Reply:
I would say more precisely that ``` toute cette arborescence est enfant du module implicite `crate` ```, instead ?

Reply:
I don't think this is mendatory, but if you think it is, then you might change it to
```markdown
le module implicite `crate` est le parent de toute cette arborescence
```
instead.

Reply:
Yeah, it's better :wink: